src/scripts/main.js

```diff
@@ -1,3 +1,58 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const thead = document.querySelector('thead');
+const tfoot = document.querySelector('tfoot');
+const theadThs = thead.querySelectorAll('th');
+const tfootThs = tfoot.querySelectorAll('th');
+
+function convNumber(text) {
+  return Number(text.replace(/[^0-9.-]+/g, ''));
+}
+
+table.addEventListener('click', (e) => {
+  const item = e.target;
+
+  for (const th of tfootThs) {
+    if (item === th) {
+      return;
+    }
+  }
+
+  if (item === theadThs[0]) {
+    const sortedRows = Array.from(table.rows)
+      .slice(1, -1)
+      .sort((rowA, rowB) =>
+        rowA.cells[0].innerHTML > rowB.cells[0].innerHTML ? 1 : -1);
+
+    table.tBodies[0].append(...sortedRows);
```

Comment:
you are doing almost the same in each case

add `switch` block inside `sort` callback to avoid duplicating code
Hint: you may join some cases if you handle them in the same way:
```
case 1:
case 2:
  // you way to handle these cases
  ```